Fix omnibus-external-postgres terraform scenario not associating vm public ip to the postgres firewall on first run

The `omnibus-external-postgres` terraform scenario currently has an issue where the `arm_instance` module does not return a value for `azurerm_public_ip.default.ip_address` on the first `apply`.

```
azurerm_postgresql_firewall_rule.default: Creating...
  name:                "" => "omnibus-external-postgres-ubuntu-1804"
  resource_group_name: "" => "csnapp-chef_server-test"
  server_name:         "" => "omnibus-external-postgres-ubuntu-1804"

Error: Error applying plan:

1 error occurred:
	* azurerm_postgresql_firewall_rule.default: 1 error occurred:
	* azurerm_postgresql_firewall_rule.default: postgresql.FirewallRulesClient#CreateOrUpdate: Invalid input: autorest/validation: validation failed: parameter=parameters.FirewallRuleProperties.StartIPAddress constraint=Pattern value="" details: value doesn't match pattern ^(([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5])\.){3}([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5])$





Terraform does not automatically rollback in the face of errors.
Instead, your Terraform state file has been partially updated with
any resources that successfully completed. Please address the error
above and apply again to incrementally change your infrastructure.


make: *** [Makefile:133: apply] Error 1
```

--- a/terraform/azure/README.md
+++ b/terraform/azure/README.md
@@ -22,11 +22,11 @@ The test scenarios require the `ARM_SSH_KEY_FILE` environment variable to be pop
 
 ### Azure Resource Manager
 
-Scenarios require that they are run against a compatible resource manager.
+Scenarios require that they are run against a compatible resource group.
 
-You may generate a compatible resource manager using the `ARM_DEPT=Eng ARM_CONTACT=csnapp make create-resource-manager` command.
+You may generate a compatible resource group using the `ARM_DEPT=Eng ARM_CONTACT=csnapp make create-resource-group` command.
 
-***NOTE:*** **There is no automatic reaping of the resource manager.  Make sure you destroy your resource manager using the** `ARM_DEPT=Eng ARM_CONTACT=csnapp make destroy-resource-manager` **command once you are finished testing scenarios and ALL server instances have been destroyed.**
+***NOTE:*** **There is no automatic reaping of the resource group.  Make sure you destroy your resource group using the** `ARM_DEPT=Eng ARM_CONTACT=csnapp make destroy-resource-group` **command once you are finished testing scenarios and ALL server instances have been destroyed.**
 
 ## Running a Scenario
 Environment variables are used to control how the scenarios are executed and can either be passed on the command line before the `make` command or set in the shell's environment (e.g. `$HOME/.bashrc`)

--- a/terraform/azure/modules/arm_instance/main.tf
+++ b/terraform/azure/modules/arm_instance/main.tf
@@ -39,8 +39,6 @@ resource "azurerm_network_security_group" "default" {
 }
 
 resource "azurerm_public_ip" "default" {
-  depends_on = [azurerm_network_security_group.default]
-
   resource_group_name = data.azurerm_resource_group.chef_resource_group.name
   location            = data.azurerm_resource_group.chef_resource_group.location
 
@@ -55,8 +53,6 @@ resource "azurerm_public_ip" "default" {
 }
 
 resource "azurerm_network_interface" "default" {
-  depends_on = [azurerm_public_ip.default]
-
   resource_group_name       = data.azurerm_resource_group.chef_resource_group.name
   location                  = data.azurerm_resource_group.chef_resource_group.location
 
@@ -76,8 +72,6 @@ resource "azurerm_network_interface" "default" {
 }
 
 resource "azurerm_virtual_machine" "default" {
-  depends_on = [azurerm_network_interface.default]
-
   resource_group_name = data.azurerm_resource_group.chef_resource_group.name
   location            = data.azurerm_resource_group.chef_resource_group.location
 
@@ -118,4 +112,13 @@ resource "azurerm_virtual_machine" "default" {
     X-Dept    = var.arm_department
     X-Contact = var.arm_contact
   }
+}
+
+# obtain the ip address after the public ip has been assigned to the virtual machine
+data "azurerm_public_ip" "default" {
+	depends_on = [azurerm_virtual_machine.default]
+
+  resource_group_name = data.azurerm_resource_group.chef_resource_group.name
+
+  name                = "${var.build_prefix}${var.name}-${local.arm_resource_group_name}"
 }

--- a/terraform/azure/modules/arm_instance/outputs.tf
+++ b/terraform/azure/modules/arm_instance/outputs.tf
@@ -7,7 +7,7 @@ output "location" {
 }
 
 output "public_ipv4_address" {
-  value = azurerm_public_ip.default.ip_address
+  value = data.azurerm_public_ip.default.ip_address
 }
 
 output "private_ipv4_address" {

